test/network/scenarios/p2p/peer_protocol_version.js

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright © 2018 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+
+'use strict';
+
+const fs = require('fs');
+const util = require('util');
+const utils = require('../../utils');
+
+const fs_writeFile = util.promisify(fs.writeFile);
+
+module.exports = function(
+	configurations,
+	network,
+	WSPORTS,
+	TOTAL_PEERS,
+	EXPECTED_TOTAL_CONNECTIONS,
+	NUMBER_OF_MONITORING_CONNECTIONS
+) {
+	const TOTAL_PEERS_LESS_ONE = TOTAL_PEERS - 1;
+	const EXPECTED_TOTAL_CONNECTIONS_AFTER_STARTING_INCOMPATIBLE_PEER =
+		(TOTAL_PEERS_LESS_ONE - 1) * TOTAL_PEERS_LESS_ONE * 2;
+	const EXPECTED_MONITORING_CONNECTIONS_AFTER_STOPPING_A_NODE =
+		NUMBER_OF_MONITORING_CONNECTIONS - 2;
+	// One of the bi-directional monitoring connections should be down so
+	// we need to subtract 2.
+
+	describe('@p2p : peer Protocol Version', () => {
+		describe('when a node with protocol = 0.0 (invalid) tries to join the network', () => {
+			const originalProtocolVersion = __testContext.config.protocolVersion;
+			let pm2NetworkConfig;
+
+			before(async () => {
+				// lisk-core linter ci task complains because it doesn't find the file, which is
+				// okay as the file is only present on lisk-core-network tests.
+				// eslint-disable-next-line import/no-dynamic-require, import/no-unresolved
+				pm2NetworkConfig = require('../../pm2.network.json');
+				pm2NetworkConfig.apps[0].env.PROTOCOL_VERSION = '0.0';
+				await fs_writeFile(
+					`${__dirname}/../../pm2.network.json`,
+					JSON.stringify(pm2NetworkConfig, null, 4)
+				);
+
+				await network.reloadNode('node_0', true, true);
+				await network.waitForBlocksOnNode('node_1', 4);
+			});
+
+			after(async () => {
```

Comment:
Please, add two more test to verify everything comes to normality after correcting back the protocol version just modified. You can see an example on the blacklisting scenario.

I would also use the node number 9 instead of number 0 because it's the only one non forging node. Otherwise you need to enable forging when you reload the node in the after block.

Reply:
Okay, will do.
Didn't know about the special case with node number 0, thanks for pointing it out.

Reply:
Implemented another test case to check whether a node with correct protocol version joins back the network with no issues. Please @diego-G let me know what you think.